core/src/acc_text/nlg/gf/generator.clj

```diff
@@ -1,20 +1,80 @@
 (ns acc-text.nlg.gf.generator
-  (:require [jsonista.core :as json]
+  (:require [acc-text.nlg.gf.grammar :as grammar]
+            [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [jsonista.core :as json]
             [org.httpkit.client :as client]))
 
 (def read-mapper (json/object-mapper {:decode-key-fn true}))
 
-(defn compile-request [grammar]
-  @(client/request {:url     (or (System/getenv "GF_ENDPOINT") "http://localhost:8001")
-                    :method  :post
-                    :headers {"Content-type" "application/json"}
-                    :body    (json/write-value-as-string {:content (reduce str grammar)})}))
+(defn ->abstract [{::grammar/keys [module flags syntax]}]
+  (format "abstract %s = {\n    flags\n        %s;\n    cat\n        %s;\n    fun\n        %s;\n}"
+          (name module)
+          (->> flags
+               (map (fn [[flag val]]
+                      (format "%s = %s" (name flag) val)))
+               (str/join ";\n        "))
+          (->> syntax
+               (mapcat :params)
+               (cons (:startcat flags))
+               (str/join ";\n        "))
+          (->> syntax
+               (map-indexed (fn [i {:keys [name params]}]
+                              (format "Function%02d : %s"
+                                      (inc i)
+                                      (str/join " -> " (-> params (vec) (conj name))))))
+               (str/join ";\n        "))))
+
+(defn ->concrete [{::grammar/keys [instance module syntax]}]
+  (format "concrete %s of %s = {\n    lincat\n        %s;\n    lin\n        %s;\n}"
+          (name instance)
+          (name module)
+          (->> syntax
+               (group-by :ret)
+               (map (fn [[ret functions]]
+                      (format "%s = {%s: %s}"
+                              (str/join ", " (map :name functions))
+                              (name (nth ret 0))
+                              (nth ret 1))))
+               (str/join ";\n        "))
+          (->> syntax
+               (map-indexed (fn [i {:keys [params ret body]}]
+                              (format "Function%02d %s= {%s = %s}"
+                                      (inc i)
```

Comment:
We get result like:
`Function03 DictionaryItem04 = {s = DictionaryItem04.s ++ "{{title}}" };`
It does work, but normally short aliases are used instead of full data type name
`Function03 d = {s = d.s ++ "{{title}}" };` which makes a lot more readable

Reply:
will deal with this in next iteration